src/mutation-observer.js

```diff
@@ -0,0 +1,87 @@
+const { observe, unobserve } = require('@nx-js/observer-util')
+const { TRAM_TAG_REACTION, TRAM_TAG_NEW_EFFECTS, TRAM_TAG_CLEANUP_EFFECTS } = require('./node-names')
+const { setup, get } = require('./namespace')
+
+/**
+ * @private
+ * @description
+ * The mutation-observer is a global instance of browsers MutationObserver
+ * which tracks when nodes are added or removed.
+ *
+ * When nodes are added we process their effects. When nodes are removed we process any cleanup,
+ * and stop observers that would trigger for that node.
+ */
+
+// process new effects for new nodes
+const processEffects = node => {
+	const hasNewEffects = node[TRAM_TAG_NEW_EFFECTS]
+
+	if (hasNewEffects) {
+		// create an array for the cleanup effects
+		node[TRAM_TAG_CLEANUP_EFFECTS] = []
+
+		// run all the effects, saving any cleanup functions to the node
+		node[TRAM_TAG_NEW_EFFECTS].forEach(effect => {
+			let cleanup
```

Comment:
ew

Reply:
this is actually an appropriate (if a bit strange), use of `let`, where it can be read before it's assigned. There are a few `let`s after the refactor, and they have to do with how the observer gets called (and it's scope).